editor/components/block-list/insertion-point.js

```diff
@@ -11,28 +17,51 @@ import {
 	getBlockInsertionPoint,
 	isBlockInsertionPointVisible,
 	getBlockCount,
+	getBlock,
+	isTyping,
 } from '../../store/selectors';
+import {
+	insertDefaultBlock,
+	startTyping,
+} from '../../store/actions';
 
-function BlockInsertionPoint( { showInsertionPoint } ) {
-	if ( ! showInsertionPoint ) {
-		return null;
-	}
+function BlockInsertionPoint( { showInsertionPoint, showInserter, index, layout, rootUID, ...props } ) {
+	const onClick = () => {
```

Comment:
For a component which is rendered for every block, I think we ought to be conscious of the referential equality of functions in render reconciliation and pre-bind these into a component class.